app/components/common/drawer/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { GlobeIcon } from '@radix-ui/react-icons';
+import moment from 'moment';
+import React from 'react';
+import { CalEvent } from '~/utils/interfaces';
+
+interface DrawerProps {
+  event: CalEvent;
+  isDrawerVisible: boolean;
+  toggleDrawer: () => void;
+}
+
+const Drawer: React.FC<DrawerProps> = ({ event, isDrawerVisible, toggleDrawer }) => (
+  <div className={`w-full h-full fixed inset-0 ${isDrawerVisible ? '' : 'invisible'}`}>
+    <div
+      onClick={toggleDrawer}
+      data-testid="drawer-background"
+      className={`w-full h-full duration-500 ease-out transition-all inset-0 absolute bg-gray-900 ${
+        isDrawerVisible ? 'opacity-50' : 'opacity-0'
+      }`}
+    ></div>
+    <div
+      onClick={toggleDrawer}
+      className={`pt-20 pb-12 px-12 w-1/4 bg-white h-full absolute right-0 duration-300 ease-out transition-all flex flex-col justify-between ${
+        isDrawerVisible ? '' : 'translate-x-full'
+      }`}
+    >
+      <div>
+        <div className="absolute cursor-pointer text-gray-600 top-0 w-8 h-8 flex items-center justify-center right-0 mt-5 mr-5">
+          <button className="w-78 h-33 left-1406 top-28 px-4 py-2 uppercase mr-8 bg-gray-100 rounded-lg ">
+            Close
+          </button>
+        </div>
+        <h1 className=" font-semibold text-3xl leading-10 capitalize">{event.title}</h1>
+        <span className="inline-flex items-center px-2.5 py-0.5 rounded-full text-sm font-medium bg-gray-100 text-gray-800 my-2">
+          <GlobeIcon style={{ color: '#737373' }} />
+          <p className="px-1">{event.type}</p>
+        </span>
```

Comment:
we already have a component `EventVisibility`. You can use it here.

Reply:
Thank you. Added it